File: lib_network/src/main/java/com/mirkowu/lib_network/ApiException.java
package com.mirkowu.lib_network;

public class ApiException extends RuntimeException {
    private final int code;
    private final String msg;


    public ApiException(int code, String msg) {
        super(msg);
        this.code = code;
        this.msg = msg;
    }

    public int code() {
        return code;
    }

    public String msg() {
        return msg;
    }
}

File: lib_network/src/main/java/com/mirkowu/lib_network/request/ApiException.kt
package com.mirkowu.lib_network.request

/**
 * api业务内部的异常
 */
open class ApiException @JvmOverloads constructor(
    val code: String,
    val msg: String?,
    val displayMsg: String? = null,
    val throwable: Throwable? = null,
) : RuntimeException(msg)